test/common_nn.py

```diff
@@ -854,6 +856,22 @@ def test_cuda(self, test_case):
                 for cpu_d_p, gpu_d_p in zip(cpu_param[1], gpu_param[1]):
                     test_case.assertEqual(cpu_d_p, gpu_d_p, 2e-4)
 
+            # Run double-backwards on CPU and GPU and compare results
+            # TODO: BatchNorm double-backwards is currently broken, see
+            # #4422; remove this when it is fixed.
+            if self.check_gradgrad and not self.get_name().startswith('test_BatchNorm'):
+                for i in range(5):
```

Comment:
Why is this running 5 times?

Reply:
I copied the pattern from the code above, which does the test for backward. It runs it 5 times because it generates random data and then sees if the CPU and GPU implementations line up; it's a probabilistic process.

Reply:
The above code does not seem like a good implementation. I don't think we get enough extra signal from running 5x to justify it. `test_nn.py` is already too slow.